### ISSUE TITLE
Point-robot-wind with sparse reward

### ISSUE THREAD

**Issue by hammer-wang** [LanqingLi1993/FOCAL-ICLR#1]
Hi,

Thanks for sharing your code!

I noticed that [here](https://github.com/LanqingLi1993/FOCAL-ICLR/blob/e5b2f59f0480ffbba648c4d05146cb1d038e3d20/rlkit/envs/non_mujoco_point_robot.py#L161) you are using dense reward for the SAC training on point-robot-wind. The reward is also dense for sparse-point-robot. 

If I switch from dense reward to sparse reward, the SAC cannot achieve the goal even after a really long training due to the lack of reward signal. This is likely caused by the hyperparameters used for dense reward are not good for the sparse setting. I'm wondering whether you could provide some advice on how to train SAC on the sparse setting. Thanks! 

Best,
Haozhu

**Comment by LanqingLi1993** [comment]
Hello Haozhu,

For FOCAL, SAC was only used as the behavior policy to generate training data, which is generally required to achieve exper-level performance. Therefore, I think there is no reason to train it with sparse reward. As a matter of fact, the training data are logged as trajectories with DENSE reward, and we could use sparsified/dense version to train our context encoder, while the policy of FOCAL was still trained with DENSE reward.  To get a sense of the data format, please refer to https://drive.google.com/file/d/1zdaUX-LC8c6AaS9We85bUvoA_9iHZcyg/view?usp=sharing for walker data we used in the paper.

Training RL algorithms in presense of sparse reward is a problem outside the scope of FOCAL. A popular work targeting sparse reward is HER: https://arxiv.org/pdf/1707.01495.pdf?source=post_page. If you are interested in coping with such problem in offline meta-RL, our latest paper might be of interest: https://arxiv.org/abs/2102.10774. Please let me know if you have any questions given the explaination.

**Comment by hammer-wang** [comment]
Hi Lanqing,

Thanks a lot for your detailed reply and pointing out the followup work. I agree with you that it's unnecessary to train the behavior policy with sparse reward. 

I'll close this issue.

Thanks again. 

Best,
Haozhu
